### ISSUE TITLE
204 No Content

### ISSUE THREAD

**Issue by deesejohn** [ts-rest/ts-rest#448]
Is there a way to omit the response body?

```ts
const contract = c.router({
  postExample: {
    method: 'POST',
    path: `/example/`,
    responses: {
      204: undefined, // Or some imported NoContentResponse
    },
    body: {
      example: z.string(),
    },
    summary: 'Example',
  },
});

const router = s.router(contract, {
  postExample: async ({ body: { example } }) => {
    await doSomething(example);
    return {
      status: 204
    };
  }
});
```

The closest I could see was:

```ts
const contract = c.router({
  postExample: {
    method: 'POST',
    path: `/example/`,
    responses: {
      204: null,
    },
    body: {
      example: z.string(),
    },
    summary: 'Example',
  },
});

const router = s.router(contract, {
  postExample: async ({ body: { example } }) => {
    await doSomething(example);
    return {
      status: 204,
      body: null
    };
  }
});
```

**Comment by eqpoqpe** [comment]
```ts
{
  responses: {
    204: c.type<undefined>(),
  },
}
```

is this ok?

**Comment by toteto** [comment]
Same is the case for having no body when making request. One has to do `client.createEmptyPost({body: undefined})`. Will try to give it a shot and figure out this TS typing issue.

**Comment by Gabrola** [comment]
Resolved in #547
